app/src/main/kotlin/com/simplemobiletools/flashlight/screens/MainScreen.kt

```diff
@@ -0,0 +1,285 @@
+            slidersSection()
+        }
+
+        Box(
+            modifier = Modifier.align(Alignment.BottomEnd),
+        ) {
+            sleepTimer()
+        }
+    }
+}
+
+@Composable
+internal fun FlashlightButton(
+    flashlightActive: Boolean,
+    onFlashlightPress: () -> Unit,
+) {
+    Icon(
+        modifier = Modifier
+            .size(AppDimensions.mainButtonSize)
+            .padding(vertical = SimpleTheme.dimens.margin.large)
+            .clickable(
+                indication = null,
+                interactionSource = rememberMutableInteractionSource(),
+                onClick = onFlashlightPress
+            ),
+        painter = painterResource(id = R.drawable.ic_flashlight_vector),
+        contentDescription = stringResource(id = R.string.flashlight_short),
+        tint = if (flashlightActive) SimpleTheme.colorScheme.primary else SimpleTheme.colorScheme.onSurface
+    )
+}
+
+@Composable
+internal fun BrightDisplayButton(
+    onBrightDisplayPress: () -> Unit,
+) {
+    Icon(
+        modifier = Modifier
+            .size(AppDimensions.smallerButtonSize)
+            .padding(vertical = SimpleTheme.dimens.margin.large)
+            .clickable(
+                indication = null,
+                interactionSource = rememberMutableInteractionSource(),
+                onClick = onBrightDisplayPress
+            ),
+        painter = painterResource(id = R.drawable.ic_bright_display_vector),
+        contentDescription = stringResource(id = R.string.bright_display),
+        tint = SimpleTheme.colorScheme.onSurface
+    )
+}
+
+@Composable
+internal fun SosButton(
+    sosActive: Boolean,
+    onSosButtonPress: () -> Unit,
+) {
+    Text(
+        modifier = Modifier
+            .padding(vertical = SimpleTheme.dimens.margin.large)
+            .clickable(
+                indication = null,
+                interactionSource = rememberMutableInteractionSource(),
+                onClick = onSosButtonPress
+            ),
+        text = stringResource(id = R.string.sos),
+        fontSize = AppDimensions.sosTextSize,
+        fontWeight = FontWeight.Bold,
+        color = if (sosActive) SimpleTheme.colorScheme.primary else SimpleTheme.colorScheme.onSurface
+    )
+}
+
+@Composable
+internal fun StroboscopeButton(
+    stroboscopeActive: Boolean,
+    onStroboscopeButtonPress: () -> Unit,
+) {
+    Icon(
+        modifier = Modifier
+            .size(AppDimensions.smallerButtonSize)
+            .padding(vertical = SimpleTheme.dimens.margin.large)
+            .clickable(
+                indication = null,
+                interactionSource = rememberMutableInteractionSource(),
+                onClick = onStroboscopeButtonPress
+            ),
+        painter = painterResource(id = R.drawable.ic_stroboscope_vector),
+        contentDescription = stringResource(id = R.string.stroboscope),
+        tint = if (stroboscopeActive) SimpleTheme.colorScheme.primary else SimpleTheme.colorScheme.onSurface
+    )
+}
```

Comment:
Kind of pointless in my opinion, since they are all different enough that the reused component would have to have a bunch of parameters (almost replicating `Icon` and `Text` components that are used for them).

Reply:
I made a shared component, but honestly, I don't think it was really needed in this case :smile: